pkg/workflows/workflow.go

```diff
@@ -566,3 +582,35 @@ func (w *workflowLifecycle) injectActiveDeadlineSeconds(wf *unstructured.Unstruc
 
 	return nil
 }
+
+func IsValidV1WorkFlow(obj runtime.Object) error {
+	wf, ok := obj.(*unstructured.Unstructured)
```

Comment:
So we no longer need to do conversion here, the `runtime.Object` has methods to check kind and apiVersion, let's remove these lines.

Reply:
341c73aa0e8f3da8125248c06684316aa805d011